### DISCUSSION TITLE
Use in Java

### DISCUSSION THREAD

**Discussion by Pey** [sceneview/sceneview#248] [Q&A]
Hi,
Thanks for your projects.
Can i use sceneview-android for Java?

**Comment by grassydragon** [comment]
Hi!
Yes, but it is more convenient to use with Kotlin so I recommend using Kotlin in new projects.

**Comment by Pey** [comment]
Thanks,
yes, kotlin is newer and better but i know java and i like by java now .

**Reply by Pey** [reply]
the future learn Kotlin and switch it

**Comment by ThomasGorisse** [comment]
SceneView v4.0.0-rc.1 is a Jetpack Compose library, which works with Kotlin. While Compose can technically be called from Java, it's not practical.

If you need to use Java, you have two options:
1. **Recommended**: Use Kotlin for the SceneView composables and keep the rest of your app in Java. Kotlin and Java interop seamlessly in the same project.
2. **Legacy**: Use an older version (0.x) which had XML-based views, though these are no longer maintained.

Most Android development has moved to Kotlin, and Compose requires it. Adding a single Kotlin file for your 3D/AR views while keeping the rest in Java is a common and well-supported pattern.
